lib/Common/Memory/HeapBlock.inl

```diff
@@ -165,25 +165,34 @@ HeapBlock::UpdateAttributesOfMarkedObjects(MarkContext * markContext, void * obj
 
     if (attributes & TrackBit)
     {
-        FinalizableObject * trackedObject = (FinalizableObject *)objectAddress;
-
-#if ENABLE_PARTIAL_GC
-        if (!markContext->GetRecycler()->inPartialCollectMode)
+#ifdef RECYCLER_VISITED_HOST
+        if (GetHeapBlockType() == HeapBlock::HeapBlockType::LargeBlockType)
```

Comment:
We could avoid adding a branch here by adding a virtual function to HeapBlock (UpdateAttributesOfTrackedMarkedObjects for example) which would also allow us to delete some code from SmallRecyclerVisitedHostHeapBlockT<TBlockAttributes>::UpdateAttributesOfMarkedObjects.

Reply:
Since the call sites know what kind of block you are operated on (i.e. LargeHeapBlock::Mark) you can just have a override on LargeHeapBlock, and call this __super version if there are code that you want to reuse here? 

---
In reply to: [183576702](https://github.com/Microsoft/ChakraCore/pull/5034#discussion_r183576702) [](ancestors = 183576702)

Reply:
Sounds good. Added UpdateAttributesOfMarkedObjects for LargeHeapBlock.

---
In reply to: [184147828](https://github.com/Microsoft/ChakraCore/pull/5034#discussion_r184147828) [](ancestors = 184147828,183576702)